adflow/pyADflow.py

```diff
@@ -2738,14 +2739,15 @@ def writeSolution(self, outputDir=None, baseName=None, number=None, writeSlices=
 
         # If we are numbering solution, it saving the sequence of
         # calls, add the call number
+        numDigits = self.getOption("writeSolutionDigits")
         if number is not None:
             # We need number based on the provided number:
-            baseName = baseName + "_%3.3d" % number
+            baseName = baseName + f"_%.{numDigits}d" % number
```

Comment:
Mixing c-style and f-string formatting looks messy. We should do either or, favoring the new f-string (like done above in this PR)